File: pset1/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int n;

    // Get input from user
    do
    {
        printf("Height: ");
        n = get_int();
    }
    // limit the height
    while (n < 0 || n > 23);

    // print the pyramid
    for (int i = 0; i < n; i++)
    {
        // print spaces on the left side
        for (int j = 0; j < n - (i + 1); j++)
        {
            printf(" ");
        }

        // print left side blocks
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }

        // print middle gap
        printf("  ");

        // print right side blocks
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }

        // go to next line
        printf("\n");
    }

    return 0;
}

File: pset6/mario.py
from cs50 import get_int

while True:
    print('Height: ', end='')
   
    n = get_int()
    
    if n >= 0 and n <= 23:
        break
    else:
        print('Height must be between 0 and 23')
        print()

for i in range(n):
    # Add left side spaces
    for j in range(n-(i+1)):
        print(' ', end='')
    
    # Add left side hashes
    for j in range(i+1):
        print('#', end='')
    
    # Add middle spaces
    print('  ', end='')
    
    # Add right side hashes
    for j in range(i+1):
        print('#', end='')
        
    print()